base/sparse/sparsematrix.jl

```diff
@@ -117,8 +117,54 @@ column. In conjunction with [`nonzeros`](@ref) and
 """
 nzrange(S::SparseMatrixCSC, col::Integer) = S.colptr[col]:(S.colptr[col+1]-1)
 
+"""
+    isvalid(S::SparseMatrixCSC; full = true) -> Bool
+
+Check if a sparse matrix `S` has a valid internal representation.
+If `full` is `false` only O(1) checks are done, otherwise full
+O(nnz) checks are performed.
+
+# Examples
+
+```jldoctest
+julia> Base.SparseArrays.isvalid(sprand(5,5,0.5))
+true
+
+julia> Base.SparseArrays.isvalid(SparseMatrixCSC(4, 4, Int[], Int[], Int[]))
+false
+```
+"""
+function Base.isvalid(S::SparseMatrixCSC; full = true)
+    length(S.colptr) != S.n + 1 && return false
```

Comment:
This check seems too strict. For explanation, please see https://github.com/JuliaLang/julia/pull/20464#issuecomment-277540516 :).